azure-servicebus/src/main/java/com/microsoft/azure/servicebus/primitives/CoreMessageSender.java

```diff
@@ -450,6 +451,7 @@ public void onError(Exception completionException)
 		}
 		else
 		{
+		    this.underlyingFactory.deregisterForConnectionError(this.sendLink);
```

Comment:
>deregisterForConnectionError [](start = 29, length = 28)

should this be common to both the if and else ? - as even if the entity is closed we need to clear the list.

Reply:
If client is closing the link, it is deregistered as soon as it is closed. This change is for the case when a connection is lost or service closes the link.

---
In reply to: [162445184](https://github.com/Azure/azure-service-bus-java/pull/180#discussion_r162445184) [](ancestors = 162445184)